src/interfaces/statics.rs

```diff
@@ -401,139 +410,262 @@ mod tests {
-        // we need mut in the commented lines below
-        #[allow(unused_mut)]
-        let mut ctx = cover_crypt.encrypt(&symmetric_key, message, authentication_data)?;
-
-        let res = cover_crypt.decrypt(&symmetric_key, &ctx, authentication_data)?;
-
-        assert_eq!(message.to_vec(), res);
+    impl EncryptionTestVector {
+        fn decrypt(&self, user_key: &str) -> Result<(), Error> {
+            let user_key = UserSecretKey::try_from_bytes(&base64::decode(user_key).unwrap())?;
+
+            let ciphertext = base64::decode(&self.ciphertext).unwrap();
+            let expected_plaintext = base64::decode(&self.plaintext).unwrap();
+
+            let header_metadata = if !self.header_metadata.is_empty() {
+                base64::decode(&self.header_metadata).unwrap()
+            } else {
+                vec![]
+            };
+
+            let authentication_data = if !self.authentication_data.is_empty() {
+                base64::decode(&self.authentication_data).unwrap()
+            } else {
+                vec![]
+            };
+            let authentication_data = if authentication_data.is_empty() {
+                None
+            } else {
+                Some(authentication_data.as_slice())
+            };
+
+            let mut de = Deserializer::new(ciphertext.as_slice());
+            let encrypted_header = EncryptedHeader::read(&mut de)?;
+            let ciphertext = de.finalize();
+            let cover_crypt = CoverCryptX25519Aes256::default();
+
+            let plaintext_header =
+                encrypted_header.decrypt(&cover_crypt, &user_key, authentication_data)?;
+            assert_eq!(plaintext_header.additional_data, header_metadata);
+            let plaintext = cover_crypt.decrypt(
+                &plaintext_header.symmetric_key,
+                &ciphertext,
+                authentication_data,
+            )?;
+            assert_eq!(expected_plaintext, plaintext);
+
+            Ok(())
+        }
 
-        //
-        // Uncomment the following code to write a new regression vector
-        //
+        pub fn new(
+            mpk: &PublicKey,
+            policy: &Policy,
+            encryption_policy: &str,
+            plaintext: &str,
+            header_metadata: Option<&[u8]>,
+            authentication_data: Option<&[u8]>,
+        ) -> Result<EncryptionTestVector, Error> {
+            let cover_crypt = CoverCryptX25519Aes256::default();
+            let (symmetric_key, encrypted_header) = EncryptedHeader::generate(
+                &cover_crypt,
+                policy,
+                mpk,
+                &AccessPolicy::from_boolean_expression(encryption_policy)?,
+                header_metadata,
+                authentication_data,
+            )?;
+
+            let mut aes_ciphertext =
+                cover_crypt.encrypt(&symmetric_key, plaintext.as_bytes(), authentication_data)?;
+            let mut ciphertext = encrypted_header.try_to_bytes()?;
```

Comment:
Maybe keep the name `encrypted_header`?
```suggestion
            let mut encrypted_header = encrypted_header.try_to_bytes()?;
```